plugins/by-name/scrollview/default.nix

```
@@ -0,0 +1,450 @@
+        The sign groups to hide in the current window for insert mode.
+        Set to `[ "all" ]` to hide all sign groups.
+
+        See `|scrollview_hide_bar_for_insert|` for hiding the scrollbar.
+      '';
+
+      signs_max_per_row = defaultNullOpts.mkInt (-1) ''
+        The maximum number of signs per row.
+        Set to `-1` to have no limit.
+      '';
+
+      signs_on_startup =
+        defaultNullOpts.mkListOf types.str
+          [
+            "diagnostics"
+            "marks"
+            "search"
+          ]
+          ''
+            The built-in sign groups to enable on startup.
+            Set to `{__empty = null;}` to disable all built-in sign groups on startup.
+            Set to `['all']` to enable all sign groups.
+            Considered only when the plugin is loaded.
+          '';
+
+      signs_overflow =
+        defaultNullOpts.mkEnumFirstDefault
+          [
+            "left"
+            "right"
+          ]
+          ''
+            The sign overflow direction (to avoid overlapping the scrollbar or other signs).
+          '';
+
+      signs_show_in_folds = defaultNullOpts.mkBool false ''
+        Whether signs on lines within hidden folds should be shown.
+        Sign groups can override this setting (e.g., the built-in cursor sign group does).
+      '';
+
+      changelist_previous_priority = mkPriorityOption "the previous item changelist sign" 15;
+
+      changelist_previous_symbol = mkNullOrStr ''
+        The symbol for the previous item changelist sign.
+
+        Defaults to an upwards arrow with tip leftwards.
+      '';
+
+      changelist_current_priority = mkPriorityOption "the current changelist sign" 10;
+
+      changelist_current_symbol = defaultNullOpts.mkStr "@" ''
+        The symbol for the current item changelist sign.
+      '';
+
+      changelist_next_priority = mkPriorityOption "the next item changelist sign" 5;
+
+      changelist_next_symbol = mkNullOrStr ''
+        The symbol for the next item changelist sign.
+
+        Defaults to a downwards arrow with tip rightwards.
+      '';
+
+      conflicts_bottom_priority = mkPriorityOption "conflict bottom signs" 80;
+
+      conflicts_bottom_symbol = defaultNullOpts.mkNullable symbolType ">" ''
+        The symbol for conflict bottom signs.
+
+        ${symbolCommonDesc}
+      '';
+
+      conflicts_middle_priority = mkPriorityOption "conflict middle signs" 75;
+
+      conflicts_middle_symbol = defaultNullOpts.mkNullable symbolType "=" ''
+        The symbol for conflict middle signs.
+
+        ${symbolCommonDesc}
+      '';
+
+      conflicts_top_priority = mkPriorityOption "conflict top signs" 70;
+
+      conflicts_top_symbol = defaultNullOpts.mkNullable symbolType "<" ''
+        The symbol for conflict top signs.
+
+        ${symbolCommonDesc}
+      '';
+
+      cursor_priority = mkPriorityOption "cursor signs" 0;
+
+      cursor_symbol = mkNullOrStr ''
```

```suggestion
      cursor_symbol = defaultNullOpts.mkStr ''
```

IIRC, mkNullOrStr doesn't inject rawLua support?

Perhaps you're using the non-defaultNullOpts helper so that you don't need to specify a pluginDefault? I wonder if it'd be better to use the `mkStr'` "prime" variant to have more flexibile arguments then?

---

> IIRC, mkNullOrStr doesn't inject rawLua support?

[I checked and it does](https://github.com/nix-community/nixvim/blob/42ea1626cb002fa759a6b1e2841bfc80a4e59615/lib/options.nix#L90).

> Perhaps you're using the non-defaultNullOpts helper so that you don't need to specify a pluginDefault?

Yes, this is the reason.



> I wonder if it'd be better to use the `mkStr'` "prime" variant to have more flexibile arguments then?

Sure, but wouldn't this be more verbose for nothing ?